### ISSUE TITLE
Expose RNG seed arguments everywhere

### ISSUE THREAD

**Issue by null-a** [pyro-ppl/brmp#11]

**Comment by null-a** [comment]
There should be a method for fixing the rng seed that works consistently across back ends. The nuts implementation of the NumPyro back end already takes a rng seed as argument. This seems ok, so we should probably add something similar to the Pyro back end.

A rng is also used by `fitted` when sampling from the posterior predictive, so this might also benefit from taking a seed as an argument.

Finally, it would to have someone confirm that the method we use to generate a default seed (for use with NumPyro) is sensible.

**Comment by neerajprad** [comment]
We recently created a `seed` handler as a generic way to fix the seed in a backend-agnostic way. So something like the following should work on either Pyro, or NumPyro:

```
with handlers.seed(1):
    # do some random computation
```

> A rng is also used by fitted when sampling from the posterior predictive, so this might also benefit from taking a seed as an argument.

I'm not sure where `rng` is used. Generally, in NumPyro, any stochastic function should take in an explicit `rng_seed` rather than relying on a seed wrapped model. This also allows it to be easily vectorized using `vmap`. The `seed` handler can be confusing, and requires a bit of care otherwise it is easy to end up reusing the same rng keys. Thankfully, users don't have to use this handler, and it is more for internal convenience! 

See below for an example which might seem obvious (these kinds of things are best avoided by seeding inside utility functions rather than passing a seeded model around): 

```python
>>> def fn():
     return numpyro.sample('n', dist.Normal(0., 1.))

>>> seedf = handlers.seed(fn, 1)

>>> seedf()
DeviceArray(-1.1470195, dtype=float32)

>>> seedf()  # different result
DeviceArray(-1.092164, dtype=float32)

>>> # All same; no randomness
>>> jax.vmap(lambda x: seedf())(np.arange(10))  
DeviceArray([1.05198181, 1.05198181, 1.05198181, 1.05198181, 1.05198181,
             1.05198181, 1.05198181, 1.05198181, 1.05198181, 1.05198181],
            dtype=float32)

>>> # Seed inside vmap
>>> jax.vmap(lambda x: handlers.seed(fn, x)())(np.arange(10))
DeviceArray([-1.25153887, -1.14701962, -0.1671523 , -0.05887513,
              0.78741592,  1.06428206, -0.93262535, -1.46220052,
             -0.10444827, -0.23851034], dtype=float32)
```

**Comment by null-a** [comment]
Thanks @neerajprad, that's helpful. I intend to address this after the next Pyro release.
